okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/HostMetricsRegistry.java

```diff
@@ -89,4 +96,49 @@ public void recordIoException(String serviceName, String hostname, int port) {
         @Value.Parameter
         int port();
     }
+
+    private final class HostEventRegistryCallback implements HostEventCallback {
+
+        private final ServiceHostAndPort target;
+        private final Supplier<DefaultHostMetrics> hostMetricsSupplier;
+
+        HostEventRegistryCallback(String serviceName, String hostname, int port) {
+            this.target = ImmutableServiceHostAndPort.of(serviceName, hostname, port);
+            // recomputed each hour, must be shorter than the 1 day timeout used by the cache.
+            // This avoids an unnecessary cache load to record metrics.
+            this.hostMetricsSupplier =
+                    Suppliers.memoizeWithExpiration(() -> hostMetrics.get(target), 1, TimeUnit.HOURS);
```

Comment:
two layers of expiry time seems a little much for me :/ we're already micro-optimising pretty hard to avoid allocation and GC here - would it really be so bad to just take the cache read?

Reply:
Since we're inside a single class here I think it should be fine. We could reduce the timeout to 1 minute and still reduce map lookups by several orders of magnitude.

Reply:
Argh ok

Reply:
Do you want this to merge as is, or with a reduced memoize duration?